### ISSUE TITLE
Problems with displaying UI messages on specific cases from version 3.1.2105.0+

### ISSUE THREAD

**Issue by apr-un** [neos-sdi/adfsmfa#189]
Hi @redhook62 
I think that I found a bug in version 3.1.2105.0

When user try to log in with MFA enabled, and won't select authentication method on form "ChooseMethodForm" but instead click Cancel (mfa-cancelButton), systems ends on form "lockingForm" with no message (empty string) and only visible Close button (buttonquit), which in fact does nothing (just reloads page). 

Instead of empty message there should be visible message from resource "ErrorInvalidIdentificationRestart" (Invalid identification, please restart your session.)

I think that empty mesage is caused by changes in BasePresentation.cs in commit 8370d52a, which adds support for custom provider. On line 165 there is called constructor AdapterPresentationDefault with only two params instead five like in earlier version - omitted are: message, suite, disableoptions - so message defaults to empty value :(

https://github.com/neos-sdi/adfsmfa/commit/8370d52ab9e04eef100ff9a6cd8401186654c271#diff-db00a0ebb157e51327183df3faae8cfd8b1e915548c00d9d0874ece17413658f

Please check that code.
Regards
Arnold

**Comment by apr-un** [comment]
Similar problems with lack of UI messages exists when user tries to change password - there is no confirmation/error when password doesn't meet complexity requirements.

**Comment by redhook62** [comment]
I don't remember, but it is possible that we have disabled these messages so as not to display too precise information.
I will look for the next update

**Comment by apr-un** [comment]
I think that this bug can have more reach - we did some more tests and there are no messages when users correctly finishes biometric registration, just Finish button :(

**Comment by redhook62** [comment]
Hi, @apr-un 

I did not notice any such errors, however this could be due to the wrong formatting of the "SameSite".
More serious, if you cancel your session ("Cancel") you bypass the MFA (version 2106 or older?)
A blocking exception is not raised ...

a quick update soon

regards

**Comment by redhook62** [comment]
Done  in 3.1.2107.0

**Comment by apr-un** [comment]
@redhook62 but fix in 2107 didn't address "no message" problem - in many places it should show something like this (with redacted logo ;))
![obraz](https://user-images.githubusercontent.com/83223821/124281975-8fb50480-db4a-11eb-8cf2-131c892c7686.png)
but it shows something like this:
![obraz](https://user-images.githubusercontent.com/83223821/124282143-c854de00-db4a-11eb-9c11-daa7fb8c73f3.png)

Empty white popup with one button is not user friendly :)

**Comment by apr-un** [comment]
@redhook62  check please here: 
https://github.com/neos-sdi/adfsmfa/blob/master/Neos.IdentityServer%203.1/Neos.IdentityServer.MultiFactor/Neos.IdentityServer.MultiFactor.BasePresentation.cs
in function
```public override string GetFormHtmlMessageZone(AuthenticationContext usercontext)
        {
            string result = string.Empty;
            if (IsPermanentFailure)
            {
                result += "<br/>";
                if (!String.IsNullOrEmpty(usercontext.UIMessage))
                    result += "<div id=\"error\" class=\"fieldMargin error smallText\"><label id=\"errorText\" name=\"errorText\" for=\"\">" + usercontext.UIMessage + "</label></div>";
                else
                    result += "<div id=\"error\" class=\"fieldMargin error smallText\"><label id=\"errorText\" name=\"errorText\" for=\"\">" + Resources.GetString(ResourcesLocaleKind.Html, "HtmlErrorRestartSession") + "</label></div>";
            }
            else
            {
                if (!String.IsNullOrEmpty(usercontext.UIMessage)) /// empty UIMessage? No info for user :(
                {
                    result += "<br/>";
                    if (IsMessage)
                        result += "<div id=\"error\" class=\"fieldMargin smallText\" style=\"color: #6FA400\"><label id=\"errorText\" name=\"errorText\" for=\"\">" + usercontext.UIMessage + "</label></div>";
                    else
                        result += "<div id=\"error\" class=\"fieldMargin error smallText\"><label id=\"errorText\" name=\"errorText\" for=\"\">" + usercontext.UIMessage + "</label></div>";  
                } /// <-- HERE, else ? show some generic error message when UIMessage is empty
            }
            return result;```

I did check that on debug and message which should be shown to user is just ignored (not set when creating Adapter) - in case of !IsPermanentFailure system always try to show usercontext.UIMessage (which is empty, because it is set in constructor with default equal to empty string).

**Comment by redhook62** [comment]
@apr-un 

As stated, I cannot reproduce your problem.
Can you indicate the full context and the constructor call ?
This code is quite old, but why not add a generic message. a message must always be passed! 
if there is no message why invent one ? 
give me more informations to reproduce.

regards

**Comment by redhook62** [comment]
Hi, @apr-un 

Yes finally ! I managed to reproduce the problem.
This only happens with the default interface (2016 / 2012r2), for the ADFS 2019 paginated interface there is no problem.

regards

**Comment by redhook62** [comment]
Hi @apr-un 

Solved ! -> new build 3.1.2107.0
